src/smt/theory_lra.cpp

```diff
@@ -2088,8 +2088,8 @@ class theory_lra::imp {
             vi = m_todo_vars.back();
             m_todo_vars.pop_back();
             lp::lar_term const& term = m_solver->get_term(vi);
-            for (auto i = term.m_coeffs.begin(); i != term.m_coeffs.end(); ++i) {
-                lp::var_index wi = i->first;
+            for (auto const& coeff : term.m_coeffs) {
+                lp::var_index wi = coeff.first;
```

Comment:
I believe we can use this more readable syntax here 
for (auto const& p : term.m_coeffs) { 
                lp::var_index wi = p.var();  // p.coeff() is also available

Reply:
add these improvements directly in a pull request into z3prover/z3.git lev.